test/DelayTest.js

```diff
@@ -1,142 +1,273 @@
+          })
+        })
 
-                    const {timestamp: expectedBlockTime} = await web3.eth.getBlock('latest')
-                    const {timeSubmitted: actualTimeSubmitted, evmCallScript: actualCallScript} = await delay.delayedScripts(0)
-                    const actualNewScriptIndex = await delay.delayedScriptsNewIndex()
+        describe('cancelExecution(uint256 _delayedScriptId)', () => {
+          beforeEach('create delayed script', async () => {
+            await delay.forward(script)
+          })
 
-                    assert.closeTo(actualTimeSubmitted.toNumber(), expectedBlockTime, 3)
-                    assert.equal(actualCallScript, script)
-                    assert.equal(actualNewScriptIndex, 1)
-                })
+          it('cancels execution script', async () => {
+            await delay.cancelExecution(0)
 
-                it('reverts when permission revoked', async () => {
-                    await daoDeployment.acl.revokePermission(rootAccount, delay.address, DELAY_EXECUTION_ROLE)
+            const {
+              executionTime: actualExecutionTime,
+              evmCallScript: actualCallScript,
+              pausedAt: actualPausedAt,
+            } = await delay.delayedScripts(0)
 
-                    const forwardReceipt =  delay.forward(script)
+            assert.equal(actualExecutionTime, 0)
+            assert.equal(actualCallScript, null)
+            assert.equal(actualPausedAt, 0)
+          })
+        })
 
-                    await assertRevert(forwardReceipt, "DELAY_CAN_NOT_FORWARD")
-                })
-            })
+        describe('execute(uint256 _delayedScriptId)', () => {
+          beforeEach('create delayed script', async () => {
+            await delay.forward(script)
+          })
+
+          it('executes the script after the delay has elapsed and deletes script', async () => {
+            await timeTravel(web3)(INITIAL_DELAY + 3)
+
+            await delay.execute(0)
+            const actualExecutionCounter = await executionTarget.counter()
+            const {
+              executionTime: actualExecutionTime,
+              evmCallScript: actualCallScript,
+              pausedAt: actualPausedAt,
+            } = await delay.delayedScripts(0)
+            assert.equal(actualExecutionCounter, 1)
+            assert.equal(actualExecutionTime, 0)
+            assert.equal(actualCallScript, null)
+            assert.equal(actualPausedAt, 0)
+          })
+
+          it('reverts when script does not exist', async () => {
+            await assertRevert(delay.execute(1), 'DELAY_NO_SCRIPT')
+          })
+
+          it('reverts when executing script before execution time', async () => {
+            await assertRevert(delay.execute(0), 'DELAY_CAN_NOT_EXECUTE')
+          })
+
+          it('reverts when executing paused script', async () => {
+            await delay.pauseExecution(0)
+            await assertRevert(delay.execute(0), 'DELAY_CAN_NOT_EXECUTE')
```

Comment:
This test would fail with this error whether you pause or not. Set it up so `delay.execute(0)` calls successfully (do the `timeTravel()` call) then put in the `pauseExecution()` to ensure it fails as expected.

Reply:
Ahh yes! Should have seen that.